src/lib/formats/sdf_dsk.cpp

```diff
@@ -0,0 +1,202 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol, tim lindner
+/*********************************************************************
+
+    formats/sdf_dsk.h
+
+    SDF disk images. Format created by Darren Atkinson for use with
+    his CoCoSDC floppy disk emulator.
+    
+    http://cocosdc.blogspot.com/p/sd-card-socket-sd-card-socket-is-push.html
+
+*********************************************************************/
+
+#include <assert.h>
+
+#include "sdf_dsk.h"
+
+
+sdf_format::sdf_format()
+{
+}
+
+
+const char *sdf_format::name() const
+{
+	return "sdf";
+}
+
+
+const char *sdf_format::description() const
+{
+	return "SDF disk image";
+}
+
+
+const char *sdf_format::extensions() const
+{
+	return "sdf";
+}
+
+
+int sdf_format::identify(io_generic *io, uint32_t form_factor)
+{
+	const int header_size = 512;
+	uint8_t header[header_size];
+
+	uint64_t size = io_generic_size(io);
+
+	if (size < header_size)
+	{
+		return 0;
+	}
+	
+	io_generic_read(io, header, 0, header_size);
+
+	int tracks = header[4];
+	int track_size = 256 + 6250 + 150; // header, track, padding
+	int heads = header[5];
+
+	// Check magic bytes
+	if (header[0] != 'S' && header[1] != 'D' && header[2] != 'F' && header[3] != '1')
+	{
+		return 0;
+	}
+
+	// Check heads
+	if (heads != 1 && heads != 2)
+	{
+		return 0;
+	}
+	
+	if (size == header_size + heads * tracks * track_size)
+	{
+		return 100;
+	}
+
+	return 0;
+}
+
+
+bool sdf_format::load(io_generic *io, uint32_t form_factor, floppy_image *image)
+{
+	const int header_size = 512;
```

Comment:
Since you're doing this constant in multiple places, you should lift it out to the top of the source file, or make it a `static constexpr` protected class member.